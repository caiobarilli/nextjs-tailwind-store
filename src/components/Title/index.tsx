import Link from 'next/link'

const Title: React.FC<{
  children: string
  size: 'sm' | 'md'
  className?: string
  href?: string
}> = ({ children, className, href, size }) => {
  switch (size) {
    case 'sm':
      if (href) {
        return (
          <Link
            href={href}
            className={`dark:text-white text-left font-bold text-gray-900 ${className !== undefined && className}`}
          >
            {children}
          </Link>
        )
      }
      return (
        <h3
          className={`dark:text-white text-left font-bold text-gray-900 ${className !== undefined && className}`}
        >
          {children}
        </h3>
      )

    case 'md':
      if (href) {
        return (
          <Link
            href={href}
            className={`dark:text-white uppercase tracking-wide no-underline hover:no-underline font-bold text-gray-800 text-xl ${className !== undefined && className}`}
          >
            {children}
          </Link>
        )
      }
      return (
        <h2
          className={`dark:text-white uppercase tracking-wide no-underline hover:no-underline font-bold text-gray-800 text-xl ${className !== undefined && className}`}
        >
          {children}
        </h2>
      )

    default:
      break
  }
}

export default Title
