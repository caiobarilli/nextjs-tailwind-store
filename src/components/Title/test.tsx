import { render, screen } from '@testing-library/react'

import Title from '.'

describe('<Title children="text" />', () => {
  it('should render the container with children', () => {
    const { container } = render(<Title size="md">Text</Title>)

    expect(screen.getByRole('heading', { name: /Text/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
