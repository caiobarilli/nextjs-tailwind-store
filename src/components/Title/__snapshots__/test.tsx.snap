// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title children="text" /> should render the container with children 1`] = `
<h2
  class="dark:text-white uppercase tracking-wide no-underline hover:no-underline font-bold text-gray-800 text-xl false"
>
  Text
</h2>
`;
